torch/CMakeLists.txt

```diff
@@ -334,6 +334,7 @@ if(OPENMP_FOUND)
     message(STATUS "Compiling with OpenMP")
   endif()
   target_compile_options(torch INTERFACE ${OpenMP_CXX_FLAGS})
+  target_link_libraries(torch -fopenmp)
```

Comment:
CC @JerryShih 

I noticed a new line on your PR which I missed previously: https://github.com/pytorch/pytorch/commit/8901935ad42fe9bf093d1106ea43606008a4024d#diff-dcf5891602b4162c36c2125c806639c5L635

Here, you unconditionally deleted the `CMAKE_EXE_LINKER_FLAGS` setting. That's probably why the PPC build is failing. Do you know why we did that? Can we add that line back?

Reply:
> CC @JerryShih
> 
> I noticed a new line on your PR which I missed previously: [8901935#diff-dcf5891602b4162c36c2125c806639c5L635](https://github.com/pytorch/pytorch/commit/8901935ad42fe9bf093d1106ea43606008a4024d#diff-dcf5891602b4162c36c2125c806639c5L635)
> 
> Here, you unconditionally deleted the `CMAKE_EXE_LINKER_FLAGS` setting. That's probably why the PPC build is failing. Do you know why we did that? Can we add that line back?

@ezyang 
I had tested with gcc and clang. It only needs to add the compile option ${OpenMP_CXX_FLAGS}. Then, we will have all the openmp compiling and linking supports.
Here is an exception for "Apple clang" compiler[1]. It needs add the linking setting explicitly. That's why I have the #14473 pr. 
[1]
https://github.com/pytorch/pytorch/commit/8901935ad42fe9bf093d1106ea43606008a4024d#diff-94eab65863273226cd6b960d685a427eR342

Reply:
@JerryShih First, I stand corrected. The line that I quoted is irrelevant and agree with your analysis in #14627 that that line doesn't do anything.

The line quoted here is different from the line above; it uses `target_link_libraries` to set the link flag, rather than `CMAKE_EXE_LINKER_FLAGS`. You observed that Apple clang needs the linking setting explicitly. It seems @avmgithub has observed another situation where the linking setting is also needed. Is there any problem with just always having it?

Yes, I think the point of this bug report, and also other reports, is that it seems the linking setting is explicitly needed in other cases too, that are not covered in our CI, but affect our users.